src/pynxtools/dataconverter/readers/multi/reader.py

```diff
@@ -0,0 +1,446 @@
+    def handle_objects(self, objects: Tuple[Any]) -> Dict[str, Any]:
+        """
+        Handles the objects passed into the reader.
+        """
+        return {}
+
+    def get_attr(self, key: str, path: str) -> Any:
+        """
+        Returns an attributes from the given path.
+        Should return None if the path does not exist.
+        """
+        return None
+
+    def get_data(self, key: str, path: str) -> Any:
+        """
+        Returns data from the given path.
+        Should return None if the path does not exist.
+        """
+        return None
+
+    def get_eln_data(self, key: str, path: str) -> Any:
+        """
+        Returns data from the given eln path.
+        Should return None if the path does not exist.
+        """
+        return {}
+
+    def get_data_dims(self, key: str, path: str) -> List[str]:
+        """
+        Returns the dimensions of the data from the given path.
+        """
+        return []
+
+    def get_entry_names(self) -> List[str]:
+        """
+        Returns a list of entry names which should be constructed from the data.
+        Defaults to creating a single entry named "entry".
+        """
+        return ["entry"]
+
+    def post_process(self) -> None:
+        """
+        Do postprocessing after all files and config file are read.
+        """
+
+    def read(
+        self,
+        template: dict = None,
+        file_paths: Tuple[str] = None,
+        objects: Tuple[Any] = None,
+        **kwargs,
+    ) -> dict:
+        """
+        Reads data from multiple files and passes them to the appropriate functions
+        in the extensions dict.
+        """
+        self.kwargs = kwargs
+        self.config_file = self.kwargs.get("config_file", self.config_file)
+        self.overwrite_keys = self.kwargs.get("overwrite_keys", self.overwrite_keys)
+
+        template = Template(overwrite_keys=self.overwrite_keys)
+
+        def get_processing_order(path: str) -> Tuple[int, Union[str, int]]:
+            """
+            Returns the processing order of the file.
+            """
+            ext = os.path.splitext(path)[1]
+            if self.processing_order is None or ext not in self.processing_order:
+                return (1, ext)
+            return (0, self.processing_order.index(ext))
+
+        sorted_paths = sorted(file_paths, key=get_processing_order)
+        for file_path in sorted_paths:
+            extension = os.path.splitext(file_path)[1].lower()
+            if extension not in self.extensions:
+                logger.warning(
+                    f"File {file_path} has an unsupported extension, ignoring file."
+                )
+                continue
+            if not os.path.exists(file_path):
+                logger.warning(f"File {file_path} does not exist, ignoring entry.")
+                continue
+
+            template.update(self.extensions.get(extension, lambda _: {})(file_path))
+
+        template.update(self.setup_template())
```

Comment:
What does this line do? For my tests, nothing. (apparently). template is still empty afterwards.

Reply:
Same as above